### PR TITLE
Deduplicating JSON objects

### LINKED ISSUE #42984: 36 occurrences where different JSON objects share same type and id
# Describe the bug

While preparing to make `tools/json_tools/util.py` work with `copy-from` I've discovered not only that there are >500 instances where same id is used on different JSON objects (which is not ideal but seems to be expected) but also that there are 36 cases where they share both `type` and `id`, which is questionable.

# Lists by status

Included in the PR:

type | id
-- | --
AMMO | oxygen
event_statistic | num_broken_bone
harvest | demihuman
json_flag | ELECTRIC_IMMUNE
ammo_effect | LIGHTNING
trait_group | Exp_Eng_Electrical_Group
MONSTER | mon_wasp_small
effect_type | magnesium
MIGRATION | l_enforcer_45
ammunition_type | oxygen
TOOL | tool_anfo_charge
TOOL | tool_anfo_charge_act
GENERIC | cable_instrument
item_group | light_reading
item_group | my_precious
item_group | mussto_windinst
item_group | mussto_stringinst
item_group | ic_merch_vending
item_group | ic_merch
mission_definition | MISSION_ISHERWOOD_JESSE_2
talk_topic | TALK_JESSE_CARLOS
talk_topic | CWH_EVACUEE_5_IDEAS1
talk_topic | TALK_LEAVE_NOW
talk_topic | TALK_TEST_SPEAKER_EFFECT_COMPOUND_SENTINEL_CONDITIONAL

@mlangsdorf ~~is fixing~~fixed these:

type | id
-- | --
vehicle_part | ~~seat~~
vehicle_part | ~~seat_leather~~
vehicle_part | ~~reclining_seat~~
vehicle_part | ~~reclining_seat_leather~~

Intentional:

type | id
-- | --
talk_topic | TALK_FRIEND_CONVERSATION

# Steps To Reproduce

```diff
diff --git a/tools/json_tools/util.py b/tools/json_tools/util.py
index de33d82..61364b5 100755
--- a/tools/json_tools/util.py
+++ b/tools/json_tools/util.py
@@ -3,7 +3,7 @@
 """

 import argparse
-from collections import Counter, OrderedDict
+from collections import Counter, OrderedDict, defaultdict
 from fnmatch import fnmatch
 import json
 import re
@@ -42,6 +42,15 @@ def import_data(json_dir=JSON_DIR, json_fmatch=JSON_FNMATCH):
                             errors.append("Problem parsing data from file %s, reason: expected a list." % json_file)
                     else:
                         data += candidates
+    all_ids = defaultdict(set)
+    for obj in data:
+        obj_id = obj.get('id')
+        obj_type = obj.get('type')
+        if obj_id and not isinstance(obj_id, list):
+            if obj_id not in all_ids[obj_type]:
+                all_ids[obj_type].add(obj_id)
+            else:
+                print(obj_type, obj_id)
     return (data, errors)

```

# Expected behavior

At least `type` and `id` together provide enough info to identify a JSON object globally. But then, there is `abstract` now too. :(

# Versions and configuration

My fork, about 40 commits behind right now.

# Additional context
 #42958 
 #42959
 https://discordapp.com/channels/598523535169945603/598535827169083403/744539312674308138

### PR DESCRIPTION
**Pull request by int-ua**
#### Summary

SUMMARY: Bugfixes "Deduplicating JSON objects that share same type and id"

#### Purpose of change

Fixes #42984

#### Describe the solution

Removing or renaming objects, see the changes.

#### Testing

TODO, requesting help with that.

### CODE REVIEW

**Comment by curstwist** [comment]
The Isherwood changes look fine.

File: data/json/items/generic/music.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   },
   {
     "type": "GENERIC",
-    "id": "cable_instrument",
+    "id": "cable_speaker",
```
Comment (Night-Pryanik) [inline data/json/items/generic/music.json:14]: You should add a migration for this.

Comment (int-ua) [reply data/json/items/generic/music.json:14]: I'm not renaming the single occurrence, there is another `cable_instrument`, I'm adding a new item by this change.

File: data/json/obsolete.json [hunk 2]
```diff
@@ -497,69 +497,6 @@
     },
     "flags": [ "BOMB", "TRADER_AVOID" ]
   },
-  {
-    "id": "tool_anfo_charge",
```
Comment (ZhilkinSerg) [inline data/json/obsolete.json:501]: This and following item were obsoleted in https://github.com/CleverRaven/Cataclysm-DDA/pull/39492, so you probably need to delete these objects from `\data\json\items\tool\explosives.json`.

Comment (int-ua) [reply data/json/obsolete.json:501]: ok, I'll bring back these and delete them in `explosives.json` a bit later

**Comment by ZhilkinSerg** [comment]
Probably worth running this script in CI in the future.

**Comment by int-ua** [comment]
Definitely. But I'm not sure what to do about `TALK_FRIEND_CONVERSATION` since it's intentional.

**Comment by jbytheway** [comment]
@mlangsdorf Can you clarify what's up with TALK_FRIEND_CONVERSATION?  How is it working with a duplicate id?  Is this something we want to retain, or should we tweak the json to achieve the same end by other means?

**Comment by I-am-Erk** [comment]
When you have two conversation objects with the same ID, it appends the extra response options together. It's an intentional design thing that lets you keep repeating options concise or have a single universal option that gets tweaked by other things

**Pull request merged**